NUAbstractModel.js

```diff
@@ -111,22 +111,22 @@ export default class NUAbstractModel extends NUObject {
         }
     }
 
-    isValid(formValues) {
+    isValid(formValues, optionalAttrs) {
         this.validationErrors.clear();
-        this.checkErrors(formValues);
+        this.checkErrors(formValues, optionalAttrs);
         return (this.validationErrors.size === 0);
     }
 
-    checkErrors(formValues) {
+    checkErrors(formValues, optionalAttrs = []) {
         const entity = this;
         entity._validators.forEach((validator, attributeName) => {
             const attrObj = this.constructor.attributeDescriptors[attributeName];
             if (Array.isArray((validator))) {
                 validator.forEach(validatorItem => {
-                    doValidate(validatorItem, entity, attrObj, formValues);
+                    doValidate(validatorItem, entity, attrObj, formValues, optionalAttrs.includes(attrObj.name));
```

Comment:
Why do we need all this? If the attribute is to be skipped then do not perform doValidate on that attribute.

Reply:
@natabal We skip only mandatory attribute validation. But if user chooses to provide a value, then other validation rules will have to still get executed (min/max length etc)

Reply:
On line 122 it seems that you validate one attribute. If the attribute to validate for the given validator is to be skipped then just skip it here. No point in passing a new attribute all along.